scripts/barclamp_mgmt_lib.rb

```diff
@@ -421,40 +421,10 @@ def bc_install_layout_1_app(from_rpm, bc_path)
   debug "Component #{component} added to Crowbar Framework.  Review #{filelist} for files created."
 end
 
-# upload the chef parts for a barclamp
-def bc_install_layout_1_chef(from_rpm, component_paths, log)
-  components = Array.new
-  component_paths.each do |component_path|
-    components.push(File.basename(component_path))
-  end
-
-  File.open(log, "a") { |f| f.puts("======== Installing chef components -- #{Time.now.strftime('%c')} ========") }
-  debug "Capturing chef install logs to #{log}"
-
-  if from_rpm
-    rpm_files = Array.new
-    components.each do |component|
-      rpm = "crowbar-#{component}"
-      debug "obtaining chef components from #{rpm} rpm"
-      rpm_files += get_rpm_file_list(rpm)
-    end
-
-    upload_cookbooks_from_rpm rpm_files, log
-    upload_data_bags_from_rpm rpm_files, log
-    upload_roles_from_rpm rpm_files, log
-  else
-    chef = File.join component_paths, 'chef'
-    cookbooks = File.join chef, 'cookbooks'
-    databags = File.join chef, 'data_bags'
-    roles = File.join chef, 'roles'
-
-    debug "obtaining chef components from #{component_paths} directory"
-    upload_cookbooks_from_dir cookbooks, ['ALL'], log
-    upload_data_bags_from_dir databags, log
-    upload_roles_from_dir roles, log
-  end
-
-  puts "Chef components for (#{components.join(", ")}) (format v1) uploaded."
+# sync the chef parts
+def bc_install_layout_1_chef
+  rails_env = ENV["RAILS_ENV"] || "production"
+  system "cd #{CROWBAR_PATH} && RAILS_ENV=#{rails_env} bin/rake chef:upload:all"
```

Comment:
Please use `su`to not run this as root. We keep having issues with rake being called as root and creating log files that crowbar then cannot write to.